src/azul/service/avro_pfb.py

```diff
@@ -384,12 +388,23 @@ def _avro_pfb_schema(azul_avro_schema: Iterable[JSON]) -> JSON:
     null_int: ['string', 'long'],
     null_str: ['string'],
     null_int_sum_sort: ['string', 'long']
+    # null_datetime is handled with a custom logical type 'avro_datetime'
 }
 
 
-def _entity_schema_recursive(field_type: FieldTypes,
+def datetime_to_string(data, *args):
+    return format_dcp2_datetime(data)
+
+
+# Fastavro doesn't support datetime objects, so a custom logical type is used
+# to tell fastaavro how to write these as strings.
+# Note: The key has the syntax "<avro_type>-<logical_type>".
+fastavro.write.LOGICAL_WRITERS['string-avro_datetime'] = datetime_to_string
```

Comment:
There should be a mention of the fact that we're omitting the corresponding reader and why we think it's safe to do so.